### PR TITLE
feat: at_commons changes for apkam onboarding

### PR DESCRIPTION
**Pull request by murali-shris**
**- What I did**
- introduced changes in at_commons for onboarding feature 
**- How I did it**
- added syntax for keys verb
- added verb builders for pkam, enroll and keys verb
- added unit tests for  pkam, enroll and keys verb builders
- added tests in syntax_test.dart to verify syntax for keys verb
- exported  pkam, enroll and keys verb builders from at_builders.dart
- modified changelog and  pubspec for publishing
**- How to verify it**
- run the unit tests

### CODE REVIEW

**Comment by xavierchanth** [comment]
I converted this to a draft, please see the migration I've done here: https://github.com/atsign-foundation/.github/issues/62

at_commons was moved to at_libraries.

A clone of this exact PR is https://github.com/atsign-foundation/at_libraries/pull/371, we should try to review and merge that instead

You will see a lot of junk on the cloned PR due to history copying, but note the commit list, and compare it to this PR.

**Comment by murali-shris** [comment]
> I converted this to a draft, please see the migration I've done here: [atsign-foundation/.github#62](https://github.com/atsign-foundation/.github/issues/62)
> 
> at_commons was moved to at_libraries.
> 
> A clone of this exact PR is [atsign-foundation/at_libraries#371](https://github.com/atsign-foundation/at_libraries/pull/371), we should try to review and merge that instead
> 
> You will see a lot of junk on the cloned PR due to history copying, but note the commit list, and compare it to this PR.

Thanks @XavierChanth

File: packages/at_commons/lib/src/verb/syntax.dart [hunk 1]
```diff
@@ -125,7 +125,20 @@ class VerbSyntax {
...
   static const noOp = r'^noop:(?<delayMillis>\d+)$';
   static const notifyRemove = r'notify:remove:(?<id>[\w\d\-\_]+)';
-  static const enroll =
-      r'^enroll:(?<operation>request|approve|deny):(?:(?:enrollmentId:(?<enrollmentId>[a-zA-Z0-9-]+))|(?:appName:(?<appName>[a-zA-Z0-9]+):)?(?:deviceName:(?<deviceName>[a-zA-Z0-9_]+):)?(?:namespaces:\[(?<namespaces>[a-zA-Z0-9;_,]+)\]:)?(?:totp:(?<totp>[0-9]+):)?(?:apkamPublicKey:(?<apkamPublicKey>.*)))?$';
+  static const enroll = r'^enroll:(?<operation>request|approve|deny):'
+      r'(?:(?:enrollmentId:(?<enrollmentId>[a-zA-Z0-9-]+))|'
+      r'(?:appName:(?<appName>[a-zA-Z0-9]+):)?'
+      r'(?:deviceName:(?<deviceName>[a-zA-Z0-9_]+):)?'
+      r'(?:namespaces:\[(?<namespaces>[a-zA-Z0-9;_,]+)\]:)?'
+      r'(?:totp:(?<totp>[0-9]+):)?(?:apkamPublicKey:(?<apkamPublicKey>.*)))?$';
```
Comment (gkc) [inline packages/at_commons/lib/src/verb/syntax.dart:133]: What is the purpose of the `(?:` at the start of the various regex fragments in the lines above?

**Review by gkc — CHANGES_REQUESTED** [review]
One question otherwise lgtm

File: packages/at_commons/lib/src/verb/syntax.dart (hunk 1 above)
Comment (murali-shris) [reply packages/at_commons/lib/src/verb/syntax.dart:133]: ?: is for non-capturing group
e.g
deviceName:pixel will capture deviceName:pixel in the group value
?:deviceName:pixel will capture only pixel in the group value.

**Comment by gkc** [comment]
Closed as this package has been moved to the at_libraries repo

This PR's replacement is in the at_libraries as https://github.com/atsign-foundation/at_libraries/pull/371

**Pull request closed without merging**